### DISCUSSION TITLE
Does argocd-repo-server need cluster-wide role?

### DISCUSSION THREAD

**Discussion by toyamagu-2021** [argoproj/argo-helm#2239] [Q&A] [answered]
It seems like we can create a cluster-wide role for argocd-repo-server.
- https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/argocd-repo-server/clusterrole.yaml

But is there any situation we use this role? The repo-server needs to access another namespace?
[In the original repo](https://github.com/argoproj/argo-cd/tree/master/manifests/cluster-rbac), there are only two cluster-wide roles for argocd-server and argocd-application-controller.

**Comment by pdrastil** [comment, marked as answer]
If I remember correctly it was used by some Helm add-on. But I agree this role is has too wide permissions and should be probably dropped for security reasons.

**Reply by toyamagu-2021** [reply]
Thanks!
I found the related PR #919 you referred to.
It might be overkill to give such a permissive role with the default value `createClusterRoles: true`, but I understood there are some use cases.
